app/src/main/java/com/android/streetworkapp/ui/image/ConfirmImageDialog.kt

```diff
@@ -164,3 +171,45 @@ fun ConfirmImageDialog(
     }
   }
 }
+
+/**
+ * Wrapper for the ConfirmImageDialog, it's setup to allow the user to upload pictures on the db.
+ */
+@Composable
+fun ConfirmImageDialogWrapper(
+    context: Context,
+    showConfirmationDialog: MutableState<Boolean>,
+    imageViewModel: ImageViewModel,
+    tempFile: File,
+    imageUri: Uri,
+    currentPark: Park,
+    currentUser: User
+) {
+  if (showConfirmationDialog.value) {
+    ConfirmImageDialog(
+        imageUri = imageUri,
+        onConfirm = {
+          imageViewModel.uploadImage(
+              context,
+              imageUri,
+              currentPark.pid,
+              currentUser.uid,
+              { onImageUploadSuccess(tempFile) },
+              { onImageUploadFailure() })
+          showConfirmationDialog.value = false
+        },
+        onCancel = {
+          showConfirmationDialog.value = false
+          if (!tempFile.delete())
+              Log.d(AddImageButtonParams.DEBUG_PREFIX, "Failed to delete cached photo file.")
+        })
+  }
+}
+
+// TODO: make UI responsive depending on success/failure
+private fun onImageUploadSuccess(file: File) {
+  if (!file.delete())
+      Log.d(AddImageButtonParams.DEBUG_PREFIX, "Failed to delete cached photo file.")
+}
+
+private fun onImageUploadFailure() {}
```

Comment:
Perhaps we could add a small comment inside this empty {}

Reply:
done